frontend/packages/core/src/button.tsx

```diff
@@ -19,7 +19,8 @@ const StyledButton = styled(MuiButton)`
   `}
 `;
 
-interface ButtonProps extends MuiButtonProps {
+interface ButtonProps
+  extends Pick<MuiButtonProps, "endIcon" | "color" | "onClick" | "size" | "type" | "variant"> {
```

Comment:
A few comments/questions:

(1)
Is this the recommended way of auditing props? Saw examples of this pattern: https://sourcegraph.lyft.net/search?q=extends+Pick%3C&patternType=literal#2

(2)
I choose props that we're currently using / plan to use:
* [endIcon](https://github.com/lyft/clutch/blob/799ea1b8314c52b9305d0061122c8708a4f93b91/frontend/packages/wizard/src/wizard.tsx#L207)
* onClick all throughout
[size](https://github.com/lyft/clutch/blob/429fa9527cf4d1386cd1117d214502a7ceaa20b8/frontend/packages/core/src/AppLayout/feedback.tsx#L24)
* [type](https://github.com/lyft/clutch/blob/fb232990d0f852f2a68886ed4d6222008e730baa/frontend/workflows/sourcecontrol/src/create-repository.tsx#L82)
* [variant](https://github.com/lyft/clutch/blob/429fa9527cf4d1386cd1117d214502a7ceaa20b8/frontend/packages/core/src/AppLayout/feedback.tsx#L24)

Are there additional ones we want to pull in?[Button API](https://material-ui.com/api/button/), [ButtonBase API](https://material-ui.com/api/button-base/)

(3)
I didn’t think I’d have to specify the `onClick` since it's a common React event. But if I didn't include it, I got this TS message `Property 'onClick' does not exist on type 'IntrinsicAttributes & ButtonProps & { children?: ReactNode; }'.`

Reply:
(1) Yupp :)
(2) I think the only ones I would add would be disabled and startIcon (purely because we support endIcon)
(3) 👍